### ISSUE TITLE
Appium server not working with Real device IOS 13.2, Appium version is 1.15.1

### ISSUE THREAD

**Issue by RamSaiMallik** [appium/appium#13614]
**Problem**

After this step "iosDriver = new IOSDriver(new URL(action.AppiumHubUrl), capabilities);". Real device and appium server are idle. The script was not executing after this particular step.

**Software**
1. IOS 13.2
2. Appium server 1.15.1
3. npm 6.12.1
4. xcode Version 11.2.1

**Code**
capabilities.setCapability("platformName" ,"iOS"); 
		capabilities.setCapability("--command-timeout", 60000);
	    capabilities.setCapability(CapabilityType.BROWSER_NAME, "Safari");
		capabilities.setCapability(MobileCapabilityType.UDID, action.UDID);
		capabilities.setCapability("automationName", "XCUITest");
		capabilities.setCapability(MobileCapabilityType.PLATFORM_VERSION, action.IOS_PLATFORM_VERSION);
		capabilities.setCapability(MobileCapabilityType.DEVICE_NAME,action.IOS_device_name);
		capabilities.setCapability("startIWDP", true);
		capabilities.setCapability("showXcodeLog", true);
		capabilities.setCapability("xcodeSigningId", "iPhone Developer");
		capabilities.setCapability("xcodeOrgId", "Narasimha K(Personal Team)");
		capabilities.setCapability("safariIgnoreFraudWarning", true);
		capabilities.setCapability("safariAllowPopups", "true");
		capabilities.setCapability("unhandledPromptBehavior", "true");
		capabilities.setCapability("noReset", true);  // for iOS only
		capabilities.setCapability("safariOpenLinksInBackground",true);
		capabilities.setCapability("nativeWebTap",true);
		capabilities.setCapability("safariIngoreFraudWarning",true);
		capabilities.setCapability("autoDismissAlerts",false);
		capabilities.setCapability("pageLoadStrategy", "none");
		capabilities.setCapability(MobileCapabilityType.NEW_COMMAND_TIMEOUT,1000);

		try {
			iosDriver = new IOSDriver(new URL(action.AppiumHubUrl), capabilities);
			driver = iosDriver;

**Log**
[Xcode] 2019-11-19 16:25:45.662518+0530 WebDriverAgentRunner-Runner[607:100318] Enqueue Failure: Failed to launch com.apple.mobilesafari: The request to open "com.apple.mobilesafari" failed.
[Xcode] The process did launch, but has since exited or crashed. /usr/local/lib/node_modules/appium/node_modules/appium-webdriveragent/WebDriverAgentRunner/UITestingUITests.m 38 1
[Xcode]     t =     1.83s         Setting up automation session
[Xcode] 
[Xcode] 2019-11-19 16:26:45.682513+0530 WebDriverAgentRunner-Runner[607:100318] Enqueue Failure: Failed to get automation session for com.apple.mobilesafari:610: Timed out while requesting automation session for com.apple.mobilesafari (pid:610). /usr/local/lib/node_modules/appium/node_modules/appium-webdriveragent/WebDriverAgentRunner/UITestingUITests.m 38 1
[Xcode] 
[Xcode]     t =    61.85s         Wait for com.apple.mobilesafari to idle
[Xcode] 
[Xcode]     t =    62.94s Find the Application 'com.apple.mobilesafari'
[Xcode] 
[Xcode]     t =    62.95s     Requesting snapshot of accessibility hierarchy for app with pid 611
[Xcode] 
[debug] [WD Proxy] Got response with status 200: {"value":{"sessionId":"E18C1E40-9170-450E-8417-ABBC21BC6670","capabilities":{"device":"iphone","browserName":"Safari","sdkVersion":"13.2","CFBundleIdentifier":"com.apple.mobilesafari"}},"sessionId":"E18C1E40-9170-450E-8417-ABBC21BC6670"}
[WD Proxy] Determined the downstream protocol as 'W3C'
[debug] [BaseDriver] Event 'wdaSessionStarted' logged at 1574161006364 (16:26:46 GMT+0530 (India Standard Time))
[debug] [BaseDriver] Event 'wdaStarted' logged at 1574161006364 (16:26:46 GMT+0530 (India Standard Time))
[XCUITest] Skipping setting of the initial display orientation. Set the "orientation" capability to either "LANDSCAPE" or "PORTRAIT", if this is an undesired behavior.
[debug] [BaseDriver] Event 'orientationSet' logged at 1574161006364 (16:26:46 GMT+0530 (India Standard Time))
[debug] [XCUITest] Waiting for initial webview
[debug] [iOS] Navigating to most recently opened webview
[debug] [iOS] Retrieving contexts and views
[debug] [XCUITest] Selecting by url: false 
[RemoteDebugger] Remote Debugger version 5.7.0
[debug] [RemoteDebugger] useNewSafari --> false
[debug] [RemoteDebugger] Checking which communication style to use (Safari on platform version '11.0')
[debug] [RemoteDebugger] Platform version equal or higher than '12.2': false
[RemoteDebugger] Setting communication protocol: using full Web Inspector protocol communication
[debug] [RemoteDebugger] Sending connection key request
[debug] [RemoteDebugger] Sending '_rpc_reportIdentifier:' message (id: 0)
[debug] [RemoteDebugger] Connected to application
[debug] [RemoteDebugger] Selecting application
[debug] [RemoteDebugger] No applications currently connected.
[debug] [XCUITest] No web frames found.
[iOS] Could not find any webviews yet, refreshing/retrying
[debug] [RemoteDebugger] Notified that new application 'PID:611' has connected
[debug] [RemoteDebugger] Found app id key 'PID:611' for bundle 'com.apple.mobilesafari'
[debug] [RemoteDebugger] Notified that new application 'PID:613' has connected
[debug] [RemoteDebugger] Current applications available:
[debug] [RemoteDebugger]     Application: 'PID:611'
[debug] [RemoteDebugger]         id: "PID:611"
[debug] [RemoteDebugger]         isProxy: false
[debug] [RemoteDebugger]         name: "Safari"
[debug] [RemoteDebugger]         bundleId: "com.apple.mobilesafari"
[debug] [RemoteDebugger]         hostId: undefined
[debug] [RemoteDebugger]         isActive: 2
[debug] [RemoteDebugger]         isAutomationEnabled: false
[debug] [RemoteDebugger]         pageArray:
[debug] [RemoteDebugger]           - id: 1
[debug] [RemoteDebugger]             title: ""
[debug] [RemoteDebugger]             url: "about:blank"
[debug] [RemoteDebugger]             isKey: false
[debug] [RemoteDebugger]     Application: 'PID:613'
[debug] [RemoteDebugger]         id: "PID:613"
[debug] [RemoteDebugger]         isProxy: true
[debug] [RemoteDebugger]         name: "com.apple.WebKit.WebContent"
[debug] [RemoteDebugger]         bundleId: "process-com.apple.WebKit.WebContent"
[debug] [RemoteDebugger]         hostId: "PID:611"
[debug] [RemoteDebugger]         isActive: 2
[debug] [RemoteDebugger]         isAutomationEnabled: false
[debug] [RemoteDebugger]         pageArray: 'Waiting for data'
[debug] [RemoteDebugger] Page changed: {
[debug] [RemoteDebugger]   "1": {
[debug] [RemoteDebugger]     "WIRTitleKey": "",
[debug] [RemoteDebugger]     "WIRTypeKey": "WIRTypeWeb",
[debug] [RemoteDebugger]     "WIRURLKey": "about:blank",
[debug] [RemoteDebugger]     "WIRPageIdentifierKey": 1
[debug] [RemoteDebugger]   }
[debug] [RemoteDebugger] }
[debug] [iOS] Remote debugger notified us of a new page listing: {"appIdKey":"611","pageArray":[{"id":1,"title":"","url":"about:blank","isKey":false}]}
[debug] [iOS] No key id found. Choosing first id from page array
[debug] [iOS] We do not appear to have window set yet, ignoring
[debug] [iOS] Retrieving contexts and views
[debug] [XCUITest] Selecting by url: false 
[debug] [RemoteDebugger] Selecting application
[debug] [RemoteDebugger] Current applications available:
[debug] [RemoteDebugger]     Application: 'PID:611'
[debug] [RemoteDebugger]         id: "PID:611"
[debug] [RemoteDebugger]         isProxy: false
[debug] [RemoteDebugger]         name: "Safari"
[debug] [RemoteDebugger]         bundleId: "com.apple.mobilesafari"
[debug] [RemoteDebugger]         hostId: undefined
[debug] [RemoteDebugger]         isActive: 2
[debug] [RemoteDebugger]         isAutomationEnabled: false
[debug] [RemoteDebugger]         pageArray:
[debug] [RemoteDebugger]           - id: 1
[debug] [RemoteDebugger]             title: ""
[debug] [RemoteDebugger]             url: "about:blank"
[debug] [RemoteDebugger]             isKey: false
[debug] [RemoteDebugger]     Application: 'PID:613'
[debug] [RemoteDebugger]         id: "PID:613"
[debug] [RemoteDebugger]         isProxy: true
[debug] [RemoteDebugger]         name: "com.apple.WebKit.WebContent"
[debug] [RemoteDebugger]         bundleId: "process-com.apple.WebKit.WebContent"
[debug] [RemoteDebugger]         hostId: "PID:611"
[debug] [RemoteDebugger]         isActive: 2
[debug] [RemoteDebugger]         isAutomationEnabled: false
[debug] [RemoteDebugger]         pageArray: 'Waiting for data'
[debug] [RemoteDebugger] Found app id key 'PID:611' for bundle 'com.apple.mobilesafari'
[debug] [RemoteDebugger] Found separate bundleId 'process-com.apple.WebKit.WebContent' acting as proxy for 'com.apple.mobilesafari', with app id 'PID:613'
[debug] [RemoteDebugger] Trying out the possible app ids: PID:611, PID:613
[debug] [RemoteDebugger] Selecting app PID:611 (try #1 of 20)
[debug] [RemoteDebugger] Sending '_rpc_forwardGetListing:' message to app 'PID:611' (id: 1)
[debug] [RemoteDebugger] Received response from send (id: 1): '["PID:611",{"1":{"WIRTitleKey":"","WIRTypeKey":"WIRTypeWeb","WIRURLKey":"about:blank","WIRPageIdentifierKey":1}}]'
[debug] [RemoteDebugger] Received page change notice for app 'PID:611' but the listing has not changed. Ignoring.
[debug] [RemoteDebugger] Sending to Web Inspector took 8ms
[debug] [RemoteDebugger] Current applications available:
[debug] [RemoteDebugger]     Application: 'PID:611'
[debug] [RemoteDebugger]         id: "PID:611"
[debug] [RemoteDebugger]         isProxy: false
[debug] [RemoteDebugger]         name: "Safari"
[debug] [RemoteDebugger]         bundleId: "com.apple.mobilesafari"
[debug] [RemoteDebugger]         hostId: undefined
[debug] [RemoteDebugger]         isActive: 2
[debug] [RemoteDebugger]         isAutomationEnabled: false
[debug] [RemoteDebugger]         pageArray:
[debug] [RemoteDebugger]           - id: 1
[debug] [RemoteDebugger]             title: ""
[debug] [RemoteDebugger]             url: "about:blank"
[debug] [RemoteDebugger]             isKey: false
[debug] [RemoteDebugger]     Application: 'PID:613'
[debug] [RemoteDebugger]         id: "PID:613"
[debug] [RemoteDebugger]         isProxy: true
[debug] [RemoteDebugger]         name: "com.apple.WebKit.WebContent"
[debug] [RemoteDebugger]         bundleId: "process-com.apple.WebKit.WebContent"
[debug] [RemoteDebugger]         hostId: "PID:611"
[debug] [RemoteDebugger]         isActive: 2
[debug] [RemoteDebugger]         isAutomationEnabled: false
[debug] [RemoteDebugger]         pageArray: 'Waiting for data'
[debug] [RemoteDebugger] Finally selecting app PID:611: [{"id":1,"title":"","url":"about:blank","isKey":false}]
[debug] [RemoteDebugger] Selected app after 14ms
[debug] [iOS] Picking webview 'WEBVIEW_611.1'
[debug] [XCUITest] Attempting to set context to 'WEBVIEW_611.1'
[debug] [RemoteDebugger] Selecting page '1' on app 'PID:611' and forwarding socket setup
[debug] [RemoteDebugger] Sending '_rpc_forwardSocketSetup:' message to app 'PID:611', page '1' (id: 2)
[debug] [RemoteDebugger] Sending to Web Inspector took 5ms
[debug] [RemoteDebugger] Sender key set
[debug] [RemoteDebugger] Sending '_rpc_forwardSocketData:' message to app 'PID:611', page '1' (id: 3)
[RemoteDebugger] Setting communication protocol: using Target-based communication
[debug] [RemoteDebugger] Target created for app 'PID:611' and page '1': {"targetId":"page-1"}
[debug] [RemoteDebugger] Sending '_rpc_forwardSocketData:' message to app 'PID:611', page '1', target 'page-1' (id: 4)
[debug] [RemoteDebugger] Current applications available:
[debug] [RemoteDebugger]     Application: 'PID:611'
[debug] [RemoteDebugger]         id: "PID:611"
[debug] [RemoteDebugger]         isProxy: false
[debug] [RemoteDebugger]         name: "Safari"
[debug] [RemoteDebugger]         bundleId: "com.apple.mobilesafari"
[debug] [RemoteDebugger]         hostId: undefined
[debug] [RemoteDebugger]         isActive: 2
[debug] [RemoteDebugger]         isAutomationEnabled: false
[debug] [RemoteDebugger]         pageArray:
[debug] [RemoteDebugger]           - id: 1
[debug] [RemoteDebugger]             title: ""
[debug] [RemoteDebugger]             url: "about:blank"
[debug] [RemoteDebugger]             isKey: true
[debug] [RemoteDebugger]     Application: 'PID:613'
[debug] [RemoteDebugger]         id: "PID:613"
[debug] [RemoteDebugger]         isProxy: true
[debug] [RemoteDebugger]         name: "com.apple.WebKit.WebContent"
[debug] [RemoteDebugger]         bundleId: "process-com.apple.WebKit.WebContent"
[debug] [RemoteDebugger]         hostId: "PID:611"
[debug] [RemoteDebugger]         isActive: 2
[debug] [RemoteDebugger]         isAutomationEnabled: false
[debug] [RemoteDebugger]         pageArray: 'Waiting for data'
[debug] [RemoteDebugger] Page changed: {
[debug] [RemoteDebugger]   "1": {
[debug] [RemoteDebugger]     "WIRTitleKey": "",
[debug] [RemoteDebugger]     "WIRTypeKey": "WIRTypeWeb",
[debug] [RemoteDebugger]     "WIRURLKey": "about:blank",
[debug] [RemoteDebugger]     "WIRPageIdentifierKey": 1,
[debug] [RemoteDebugger]     "WIRConnectionIdentifierKey": "0d57a207-7f98-455c-ae66-be83fd75b895"
[debug] [RemoteDebugger]   }
[debug] [RemoteDebugger] }
[debug] [iOS] Remote debugger notified us of a new page listing: {"appIdKey":"611","pageArray":[{"id":1,"title":"","url":"about:blank","isKey":true}]}
[debug] [iOS] We do not appear to have window set yet, ignoring

**Comment by mykola-mokhnach** [comment]
Please try appium@beta

**Comment by RamSaiMallik** [comment]
@mykola-mokhnach 

Tried latest beta version 1.16.0-beta.1. In beta getting issue "Could not proxy command to remote server / Error: connect ECONNREFUSED"

**Comment by mykola-mokhnach** [comment]
Please provide the full log

**Comment by RamSaiMallik** [comment]
[debug] [MJSONWP (b957122f)] Calling AppiumDriver.getScreenshot() with args: ["b957122f-7cf1-4c03-b90b-baf0f8457b87"]
[debug] [XCUITest] Executing command 'getScreenshot'
[debug] [XCUITest] Taking screenshot with WDA
[debug] [WD Proxy] Matched '/screenshot' to command name 'getScreenshot'
[debug] [WD Proxy] Proxying [GET /screenshot] to [GET http://localhost:8100/session/FCDA47E2-7AC9-4E76-B3EB-579DAC4CB221/screenshot] with no body
[WD Proxy] Got an unexpected response with status undefined: {"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":8100}
[XCUITest] Error getting screenshot: An unknown server-side error occurred while processing the command. Original error: Could not proxy command to remote server. Original error: Error: connect ECONNREFUSED 127.0.0.1:8100
[debug] [XCUITest] Taking screenshot with WDA
[debug] [WD Proxy] Matched '/screenshot' to command name 'getScreenshot'
[debug] [WD Proxy] Proxying [GET /screenshot] to [GET http://localhost:8100/session/FCDA47E2-7AC9-4E76-B3EB-579DAC4CB221/screenshot] with no body
[WD Proxy] Got an unexpected response with status undefined: {"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":8100}
[debug] [XCUITest] Taking screenshot with WDA
[debug] [WD Proxy] Matched '/screenshot' to command name 'getScreenshot'
[debug] [WD Proxy] Proxying [GET /screenshot] to [GET http://localhost:8100/session/FCDA47E2-7AC9-4E76-B3EB-579DAC4CB221/screenshot] with no body
[WD Proxy] Got an unexpected response with status undefined: {"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":8100}
[debug] [MJSONWP (b957122f)] Encountered internal error running command: UnknownError: An unknown server-side error occurred while processing the command. Original error: Could not proxy command to remote server. Original error: Error: connect ECONNREFUSED 127.0.0.1:8100
[debug] [MJSONWP (b957122f)]     at JWProxy.command (/usr/local/lib/node_modules/appium/node_modules/appium-base-driver/lib/jsonwp-proxy/proxy.js:270:13)
[HTTP] <-- GET /wd/hub/session/b957122f-7cf1-4c03-b90b-baf0f8457b87/screenshot 500 1559 ms - 271
[HTTP] 
[HTTP] --> GET /wd/hub/session/b957122f-7cf1-4c03-b90b-baf0f8457b87/screenshot
[HTTP] {}
[debug] [MJSONWP (b957122f)] Calling AppiumDriver.getScreenshot() with args: ["b957122f-7cf1-4c03-b90b-baf0f8457b87"]
[debug] [XCUITest] Executing command 'getScreenshot'
[debug] [XCUITest] Taking screenshot with WDA
[debug] [WD Proxy] Matched '/screenshot' to command name 'getScreenshot'
[debug] [WD Proxy] Proxying [GET /screenshot] to [GET http://localhost:8100/session/FCDA47E2-7AC9-4E76-B3EB-579DAC4CB221/screenshot] with no body
[WD Proxy] Got an unexpected response with status undefined: {"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":8100}
[XCUITest] Error getting screenshot: An unknown server-side error occurred while processing the command. Original error: Could not proxy command to remote server. Original error: Error: connect ECONNREFUSED 127.0.0.1:8100
[debug] [XCUITest] Taking screenshot with WDA
[debug] [WD Proxy] Matched '/screenshot' to command name 'getScreenshot'
[debug] [WD Proxy] Proxying [GET /screenshot] to [GET http://localhost:8100/session/FCDA47E2-7AC9-4E76-B3EB-579DAC4CB221/screenshot] with no body
[WD Proxy] Got an unexpected response with status undefined: {"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":8100}
[debug] [XCUITest] Taking screenshot with WDA
[debug] [WD Proxy] Matched '/screenshot' to command name 'getScreenshot'
[debug] [WD Proxy] Proxying [GET /screenshot] to [GET http://localhost:8100/session/FCDA47E2-7AC9-4E76-B3EB-579DAC4CB221/screenshot] with no body
[WD Proxy] Got an unexpected response with status undefined: {"errno":"ECONNREFUSED","code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":8100}
[debug] [MJSONWP (b957122f)] Encountered internal error running command: UnknownError: An unknown server-side error occurred while processing the command. Original error: Could not proxy command to remote server. Original error: Error: connect ECONNREFUSED 127.0.0.1:8100
[debug] [MJSONWP (b957122f)]     at JWProxy.command (/usr/local/lib/node_modules/appium/node_modules/appium-base-driver/lib/jsonwp-proxy/proxy.js:270:13)
[HTTP] <-- GET /wd/hub/session/b957122f-7cf1-4c03-b90b-baf0f8457b87/screenshot 500 1018 ms - 271
[HTTP]

**Comment by mykola-mokhnach** [comment]
Sorry, but this part is not very helpful. We need the complete log as a gist

**Comment by RamSaiMallik** [comment]
https://github.com/appium/appium/blob/d7a18534fa888f2fb808abcbf06b8022afedd44f/ErrorLog

**Comment by mykola-mokhnach** [comment]
```
[BaseDriver] Shutting down because we waited 1000 seconds for a command
[DevCon Factory] Releasing connections for 81a9af36847bcec289533772a3c89dfbd45cd572 device on any port number
[DevCon Factory] Found cached connections to release: ["81a9af36847bcec289533772a3c89dfbd45cd572:8100"]
[DevCon Factory] Releasing the listener for '81a9af36847bcec289533772a3c89dfbd45cd572:8100'
[Appium] Closing session, cause was 'New Command Timeout of 1000 seconds expired. Try customizing the timeout using the 'newCommandTimeout' desired capability'
[Appium] Removing session 1ec1f209-96de-4d80-b390-e7c8792adb5c from our master session list
[debug] [DevCon Factory] Cached connections count: 0
[debug] [XCUITest] Not clearing log files. Use `clearSystemFiles` capability to turn on.
[debug] [XCUITest] In a web session. Removing remote debugger
[debug] [RemoteDebugger] Disconnecting from remote debugger
[debug] [RemoteDebugger] Cleaning up listeners
```

Before in the log I see that a session for the same device has been expired and thus all connections to the device under test have been terminated. Please check the other parallel sessions in your client code either don't use the same device or are closed properly.

**Comment by RamSaiMallik** [comment]
@mykola-mokhnach 
Used same session to execute scripts. But that not a "error" point.

**Comment by mykola-mokhnach** [comment]
> @mykola-mokhnach
> Used same session to execute scripts. But that not a "error" point.

Sorry, I don't understand your answer

**Comment by mykola-mokhnach** [comment]
Closed because of no response
